File: compute_time.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

#include <time.h>

int main()
{
	clock_t t1, t2;				// variables for computing clocks 
	double a=1.234, b=5.678;
	double T1, T2;
	int i, j, k, N=100000000;

	/* Addition and Subtraction x 3 + 1 loop */
	t1 = clock();
	for(i=0;i<N;++i)
	{
		a = a+b;
		b = a-b;
		a = a-b;
	}
	t2 = clock();
	T1 = (t2-t1)/(double) CLOCKS_PER_SEC;
//	printf("(+,-) time x 3 + 1 loop:%f\n",T1);
//	printf("(a,b)=%f %f\n", a,b);
	
	/* Addition and Subtraction x 6 + 1 loop*/
	t1 = clock();
	for(i=0;i<N;++i)
	{
		a = a+b;
		b = a-b;
		a = a-b;
		a = a+b;
		b = a-b;
		a = a-b;
	}
	t2 = clock();
	T2 = (t2-t1)/(double) CLOCKS_PER_SEC;
//	printf("(+,-) time x 6 + 1 loop:%f\n",T2);
	printf("Real (+,-) time: %f\n",(T2-T1)/3.0);	
//	printf("(a,b)=%f %f\n", a,b);

	/* Multiplication & Division x 3 + 1 loop */
	t1 = clock();
	for(i=0;i<N;++i)
	{
		a = a*b;
		b = a/b;
		a = a/b;
	}
	t2 = clock();
	T1 = (t2-t1)/(double) CLOCKS_PER_SEC;
//	printf("(*,/) time x 3 + 1 loop:%f\n",T1);
//	printf("(a,b)=%f %f\n", a,b);

	/* Multiplication & Division x 6 + 1 loop */
	t1 = clock();
	for(i=0;i<N;++i)
	{
		a = a*b;
		b = a/b;
		a = a/b;
		a = a*b;
		b = a/b;
		a = a/b;
	}
	t2 = clock();
	T2 = (t2-t1)/(double) CLOCKS_PER_SEC;
//	printf("(*,/) time x 6 + 1 loop:%f\n",T2);
	printf("Real (*,/) time: %f\n",(T2-T1)/3.0);	
//	printf("(a,b)=%f %f\n", a,b);

	/* sin x 1 + 1 loop */
	t1 = clock();
	for(i=0;i<N;++i)
	{
		a = sin(a);
	}
	t2 = clock();
	T1 = (t2-t1)/(double) CLOCKS_PER_SEC;
//	printf("(sin) time + 1 loop:%f\n",T1);
//	printf("(a,b)=%f %f\n", a,b);

	/* sin x 2 + 1 loop */
	t1 = clock();
	for(i=0;i<N;++i)
	{
		a = sin(a);
		b = sin(b);
	}
	t2 = clock();
	T2 = (t2-t1)/(double) CLOCKS_PER_SEC;
//	printf("(sin) time x 2 + 1 loop:%f\n",T2);
	printf("(sin) time: %f\n",T2-T1);
//	printf("(a,b)=%f %f\n", a,b);

	return 0;
} 

File: compute_time.m
a = 1.234; b=2.345;
tic;
for i=1:100000000
    a = a+b;
    b = a-b;
    b = a-b;
end
T1 = toc;
%fprintf('(+,-) time:%f\n',T1);
tic;
for i=1:100000000
    a = a+b;
    b = a-b;
    b = a-b;
    a = a+b;
    b = a-b;
    b = a-b;
end
T2 = toc;
%fprintf('(+,-) time:%f\n',T2);
fprintf('(+,-) time:%f\n',(T2-T1)/3.0);

tic;
for i=1:100000000
    a = a*b;
    b = a/b;
    b = a/b;
end
T1 = toc;
%fprintf('(*,/) time:%f\n',T1);
tic;
for i=1:100000000
    a = a*b;
    b = a/b;
    b = a/b;
    a = a*b;
    b = a/b;
    b = a/b;

end
T2 = toc;
%fprintf('(*,/) time:%f\n',T2);
fprintf('(*,/) time:%f\n',(T2-T1)/3.0);
tic;
for i=1:100000000
    a = sin(a);
end
T1 = toc;
tic;
for i=1:100000000
    a = sin(b);
    b = sin(a);
end
T2 = toc;
fprintf('(sin) time:%f\n', T2-T1);

tic;
for i=1:10000
    a = randi(32768,100);
end
fprintf('randi time:%f\n',toc);